src/main/java/de/hpi/bpt/argos/api/response/ResponseFactoryImpl.java

```diff
@@ -102,15 +106,54 @@ public String getProduct(long productId) {
 	 * {@inheritDoc}
 	 */
 	@Override
-	public String getAllEventTypes(long productId) {
+	public String getProductConfiguration(long productConfigurationId) {
+		ProductConfiguration configuration = entityManager.getProductConfiguration(productConfigurationId);
+
+		if (configuration == null) {
+			halt(ResponseFactory.HTTP_NOT_FOUND_CODE, "cannot find product configuration");
+		}
+
+		JsonObject jsonConfiguration = getProductConfigurationBase(configuration);
+
+		return serializer.toJson(jsonConfiguration);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getAllProductEventTypes(long productId) {
 
 		Product product = entityManager.getProduct(productId);
 
 		if (product == null) {
 			halt(ResponseFactory.HTTP_NOT_FOUND_CODE, "cannot find product");
 		}
 
-		Map<EventType, Integer> eventTypes = entityManager.getEventTypes(productId);
+		Map<EventType, Integer> eventTypes = entityManager.getProductEventTypes(productId);
+
+		JsonArray jsonEventTypes = new JsonArray();
+
+		for (Map.Entry<EventType, Integer> eventTypeEntry : eventTypes.entrySet()) {
+			JsonObject jsonEventType = getEventType(eventTypeEntry.getKey());
+			jsonEventType.addProperty("numberOfEvents", eventTypeEntry.getValue());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "numberOfEvents" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)